### PR TITLE
feat: WASM reliability improvements - stack overflow detection, memory limits, and timeout handling

### PR DESCRIPTION
**Pull request by c4milo**
## Summary

This PR adds reliability and safety features for running untrusted JavaScript code in WASM environments. It addresses four major issues:

1. **Stack overflow detection** - Fast frame-depth based detection (~0.2ms instead of 50ms WASM crash)
2. **Memory limits** - Per-instance memory limits to prevent hang-ups from large allocations
3. **Timeout handling** - Working execution timeouts with millisecond precision
4. **Misc:** - Fixes crash when JavaScript null/undefined values are passed to Go
variadic functions

> [!CAUTION]
> Changes the default `max_depth` from 1000 to 256 frames to prevent WASM corruption.

The WASM memory layout with `--stack-first` + 20 pages (1,280KB) creates a hard boundary where the C stack collides with QuickJS heap at exactly **380 frames**. Beyond this point:
- VM becomes permanently corrupted
- Simple operations like `2+2` fail
- Even `Close()` panics

## Testing
- Unit tests for MemoryLimit feature (5 test functions)
- Sandbox tests for DisableFilesystem/DisableSystemTime
- Stack overflow detection tests (sandbox_test.go, utils_test.go, value_test.go)
- Fixed edge case tests
- Threshold testing validated the 379KB/380KB corruption boundary

---

## Breaking Changes

**None.** All new features are opt-in via configuration options.

## Performance

| Feature | Performance | Improvement |
|---------|-------------|-------------|
| Stack overflow detection | ~0.2ms | 250x faster than WASM crash |
| Memory limit | ~125µs | Prevents hangs |
| Timeout handler | Millisecond precision | Accurate execution control |
| Default 256 frame limit | <1ms exception | Safe and sufficient |

---

<details>
<summary><b>WASM runtime corruption analysis</b></summary>

### WASM Memory Layout

```
┌──────────────┬────────────┬──────────────┐
│ C Stack →    │   Heap  →  │  Other Data  │
│ (0 - 380KB)  │            │              │
└──────────────┴────────────┴──────────────┘
       ↑ Stack grows right, CORRUPTS heap at ~380KB, Wazero does not implement guard pages to avoid CGO.
```

### Test Results

| MaxStackSize | Frames | Result | VM Usable | Status |
|-------------|--------|--------|-----------|---------|
| 10KB | 8 | ✅ | Yes | Clean exception |
| 50KB | 48 | ✅ | Yes | Clean exception |
| 100KB | 98 | ✅ | Yes | Clean exception |
| 200KB | 198 | ✅ | Yes | Clean exception |
| **256KB** | **254** | **✅** | **Yes** | **SAFE DEFAULT** |
| 360KB | 358 | ✅ | Yes | Clean exception |
| **379KB** | **377** | **✅** | **Yes** | **MAXIMUM SAFE** |
| **380KB** | **378** | **❌** | **No** | **CORRUPTION** |
| 500KB | 498 | ❌ | No | Corruption |
| 1MB | ~1000 | ❌ | No | Corruption |

### Root Cause

The corruption at 380KB is caused by:
1. `--stack-first` linker option places C stack at byte 0
2. Initial memory: 20 pages = 1,280KB total
3. QuickJS heap starts around byte 380,000
4. C stack collision with heap → "out of bounds memory access" → permanent corruption

Quick reproducer: https://gist.github.com/c4milo/c425589057e273163435f12d69a6bc6a

</details>

### CODE REVIEW

**Comment by c4milo** [comment]
👋🏻 @ngocphuongnb, please let me know if you'd like me to break this into multiple PRs or remove anything you may not like.

**Comment by ngocphuongnb** [comment]
@c4milo Hi there, I am very sorry for the late reply; I have been quite busy these past few days.
Regarding your question, let's keep it as one PR for the moment. I will follow up with specific suggestions if I feel it needs to be split or if any section needs removal after my review is complete. Thanks again for your contribution.

**Comment by danslo** [comment]
@ngocphuongnb Totally understand if you're busy. Is there anything we can help with to move this forward? Being able to set timeouts seems like a key feature :)

**Comment by danslo** [comment]
Gentle bump here, can we help in any way to get this PR moving?

File: qjswasm/helpers.c [hunk 1]
```diff
@@ -59,43 +60,58 @@ JSValue QJS_ThrowInternalError(JSContext *ctx, const char *fmt)
...
+    {
+        return 0;
+    }
+
+    uint64_t now_ms = get_time_ms();
+    uint64_t elapsed_ms = now_ms - start_ms;
+
+    if (elapsed_ms > timeout_ms)
+    {
+        free(ts);
```
Comment (crackcomm) [inline qjswasm/helpers.c:87]: Is this a leak if the script finishes successfully?
